cachi2/core/package_managers/yarn_classic/resolver.py

```diff
@@ -325,8 +325,10 @@ def resolve_packages(project: Project) -> Iterable[YarnClassicPackage]:
     yarn_lock = YarnLock.from_file(project.source_dir.join_within_root("yarn.lock"))
     runtime_deps = find_runtime_deps(project.package_json, yarn_lock, workspaces)
 
-    return chain(
-        [_get_main_package(project.package_json)],
-        _get_workspace_packages(project.source_dir, workspaces),
-        _get_packages_from_lockfile(project.source_dir, yarn_lock, runtime_deps),
+    return list(
+        chain(
```

Comment:
mypy seems to complain about this, I found this could work:
```
    result: list[YarnClassicPackage] = []

    result.append(_get_main_package(project.package_json))
    result.extend(_get_workspace_packages(project.source_dir, workspaces))
    result.extend(
        _get_packages_from_lockfile(project.source_dir, mirror_dir, yarn_lock, runtime_deps)
    )
    return result
```

Reply:
`mypy` is at it again. Let's see. In the proposed amendment the first call returns `WorkspacePackage` which is then gets wrapped into a `list`. The second call returns `list[WorkspacePackage]`. The third call returns `list[YarnClassicPackage]`. The function is supposed to return an `Iterable` over `YarnClassicPackage`s. By definition `YarnClassicPackage` is a union containing `WorkspacePackage`, i.e. a wider type which could be constructed from some combination of narrower ones. A list is an interable. Concating a list of narrower types to a list of wider type should result in a list of wider type.

More importantly `list(chain(...))` does exactly the same. I wonder why `mypy` decides that one way of conversion is fine from typing perspective while another one is not. I am personally not a fan of adding extra hints just to pacify mypy.